swift/StableDiffusion/pipeline/StableDiffusionPipeline.swift

```diff
@@ -261,15 +264,21 @@ public struct StableDiffusionPipeline: ResourceManaging {
                     timeStep: t,
                     sample: latents[i]
                 )
+
+                if let denoisedLatent = scheduler[i].modelOutputs.last {
+                    denoisedLatents.append(denoisedLatent)
+                }
             }
 
+            let currentLatentSamples = config.useDenoisedIntermediates ? denoisedLatents : latents
+
             // Report progress
             let progress = Progress(
                 pipeline: self,
                 prompt: config.prompt,
                 step: step,
                 stepCount: timeSteps.count,
-                currentLatentSamples: latents,
+                currentLatentSamples: currentLatentSamples,
                 configuration: config
             )
```

Comment:
@atiorh Wasn't sure if it was worth passing both latents into progress, so keeping the interface the same and just swapping out whats being passed here based on the config value.